VideoAPI/VideoApi.Events/Handlers/TransferEventHandler.cs

```diff
@@ -29,24 +29,31 @@ protected override async Task PublishStatusAsync(CallbackEvent callbackEvent)
 
             var command =
                 new UpdateParticipantStatusAndRoomCommand(SourceConference.Id, SourceParticipant.Id, participantStatus,
-                    callbackEvent.TransferTo);
+                    callbackEvent.TransferTo, callbackEvent.TransferredToRoomLabel);
             await CommandHandler.Handle(command);
 
-            if (participantStatus == ParticipantState.InConsultation)
+            if (participantStatus == ParticipantState.InConsultation && callbackEvent.TransferTo.HasValue)
             {
                 _roomReservationService.RemoveRoomReservation(SourceConference.Id, (RoomType)callbackEvent.TransferTo);
             }
         }
 
         private static ParticipantState DeriveParticipantStatusForTransferEvent(CallbackEvent callbackEvent)
```

Comment:
why static?

Reply:
it was the default from resharper/rider. Let me remove